src/utils/confighandler.cpp:517: possible use of wrong operator ?

### Flameshot Version

11.0.0

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

fedora linux

### Description

New C compiler clang-14 says:

src/utils/confighandler.cpp:517:12: warning: use of bitwise '&' with boolean operands [-Wbitwise-instead-of-logical]

Source code is

    return checkUnrecognizedSettings(log) & checkShortcutConflicts(log) &
           checkSemantics(log);

Maybe better code:

    return checkUnrecognizedSettings(log) && checkShortcutConflicts(log) &&
           checkSemantics(log);



### Steps to reproduce

_No response_

### Screenshots or screen recordings

_No response_

### System Information

fedora 36 beta

--- a/src/utils/confighandler.cpp
+++ b/src/utils/confighandler.cpp
@@ -518,7 +518,7 @@ QSet<QString> ConfigHandler::keysFromGroup(const QString& group) const
 
 bool ConfigHandler::checkForErrors(AbstractLogger* log) const
 {
-    return checkUnrecognizedSettings(log) & checkShortcutConflicts(log) &
+    return checkUnrecognizedSettings(log) && checkShortcutConflicts(log) &&
            checkSemantics(log);
 }
 

